pkg/didcomm/protocol/introduce/service.go

```diff
@@ -348,43 +306,76 @@ func (s *Service) sendMsgEvents(msg *service.StateMsg) {
 }
 
 // newDIDCommActionMsg creates new DIDCommAction message
-func (s *Service) newDIDCommActionMsg(msg *metaData) service.DIDCommAction {
+func (s *Service) newDIDCommActionMsg(md *metaData) service.DIDCommAction {
 	// create the message for the channel
 	// trigger the registered action event
 	actionStop := func(err error) {
-		// if introducee received Proposal disapprove must be true
-		if msg.Msg.Type() == ProposalMsgType {
-			msg.disapprove = true
+		// if introducee received Proposal rejected must be true
+		if md.Msg.Type() == ProposalMsgType {
+			md.rejected = true
 		}
 
-		msg.err = err
-		s.processCallback(msg)
+		md.err = err
+		s.processCallback(md)
 	}
 
 	return service.DIDCommAction{
 		ProtocolName: Introduce,
-		Message:      msg.copyMsg,
-		Continue: func(args interface{}) {
-			// there is no way to receive another interface
-			if dep, ok := args.(InvitationEnvelope); ok {
-				msg.dependency = dep
-				s.processCallback(msg)
-				return
+		Message:      md.msgClone,
+		Continue: func(opt interface{}) {
+			if fn, ok := opt.(Opt); ok {
+				fn(md.Msg.Metadata())
+			}
+
+			if md.Msg.Type() == RequestMsgType {
+				if md.Msg.Metadata()[metaRecipients] == nil {
+					md.err = errors.New("no recipients")
+				}
 			}
-			// sets an error to the message
-			actionStop(errMissingDependency)
+
+			s.processCallback(md)
 		},
-		Stop: func(err error) { actionStop(&customError{error: err}) },
+		Stop: func(err error) { actionStop(customError{error: err}) },
 	}
 }
 
+// Continue allows proceeding with the action by the psmID
+func (s *Service) Continue(psmID string, opt Opt) error {
```

Comment:
Are we planning to have explicit APIs in client package? As a consumer of the framework it would be hard to understand what Continue does in async case (am I continuing from state a -> state b or state c -> state d). I feel it would be better to have explicit/clear APIs for such cases. Internally we can use this continue function.

Reply:
For now, we do not have an API. The introduce client might provide these functions based on the `Continue` function.